apps/schema/static/schema/components/shipmentAccessRequests/dataTypes.yaml

```diff
@@ -0,0 +1,87 @@
+uuid: &uuid
+  $ref: '../core/dataTypes.yaml#/uuid'
+
+requesterId:
+  properties:
+    requester_id:
+      <<: *uuid
+      title: requester_id
+      description: User UUID of the requester
+
+approvedBy:
+  properties:
+    approved_by:
+      <<: *uuid
+      title: approved_by
+      description: User UUID of the approver
+
+approved:
+  properties:
+    approved:
+      title: approved
+      description: Whether the access request is approved, denied, or pending (null)
+      type: boolean
+      example: true
+
+approvedAt:
+  properties:
+    approved_at:
+      title: approved_at
+      description: Date/time of request approval
+      type: string
+      format: date-time
+      example: '2018-06-18T14:54:56.652732Z'
+
+shipmentPermission:
+  properties:
+    shipment_permission:
+      title: shipment_permission
+      description: Permission level for shipment endpoint access
+      type: string
```

Comment:
Maybe it would make sense to move part of the permissions description to it's own part like with `uuid` above.
Would look somewhat like:
```
permissionLevels: &permissionLevels
      type: string
      enum: [ NONE, READ_ONLY, READ_WRITE ]
      example: READ_ONLY
```
And you could reference it here with `<<: *permissionLevels`

Reply:
This could dry things up - but still couldn't be used for tracking_permission/telemetry_permission. Also you'd miss the 'endpoint' part of the description if you made it generic: Permission level for **shipment** endpoint access

I'll take a closer look and see how I feel about the change after playing with it.

Reply:
You can have the base `permissionLevels` and then overwrite the `enum` for tracking/telemetry. And I wasn't saying move the description to the `permissionLevels` part. If you look at how `uuid` does it has the `description` but that gets overwritten in everything that uses it.